### PR TITLE
WINDUP-2407 (MA) Prototype - Automated Build Process - Travis CI

### PR DESCRIPTION
**Pull request by carlosthe19916**
Created:
- .travis.yml
- sonar-project.properties

Changed:
- pom.xml
 Moved fabric8-maven-plugin to docker profile because it is trying to download a docker image from Red Hat registry which is not allowed for travis unless I configure docker credentials before.

### CODE REVIEW

**Comment by carlosthe19916** [comment]
@mrizzi I updated this PR:
- Added Travis and a Codecov badged to the README file
- Include Jacoco maven plugin into a new profile called "coverage"
- Added lombok.config file to avoid scanning of auto-generated classes 

I also removed SonarCloud integration... Related to SonarCloud these are the options I think we have:
- Use SonarCloud just into our master branch. This won't allow us to analyze PRs but we will still be able to analyze the master branch.
- Change our SonarCloud permissions configuration to allow anybody to execute an analysis. Of course, this has its disadvantages.
- Discard completely the use of SonarCloud and work just with Codecov for simple coverage reports.

What do you think?

File: pom.xml [hunk 1]
```diff
@@ -229,22 +229,80 @@
...
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>docker</id>
```
Comment (mrizzi) [inline pom.xml]:
```suggestion
            <id>fuse-fabric8-docker</id>
```

**Review by mrizzi — COMMENTED** [review]
Just one little change to the profile name

**Comment by mrizzi** [comment]
Thanks @carlosthe19916 for proposing different solutions.
I think we should adopt your first proposal:

> Use SonarCloud just into our master branch. This won't allow us to analyze PRs but we will still be able to analyze the master branch.

**Review by mrizzi — APPROVED** [review]
Thanks @carlosthe19916, the last changes look good :+1:

**Pull request merged**